drivers/pwm/pwm_mcux.c

```diff
@@ -192,7 +192,7 @@ static const struct pwm_driver_api pwm_mcux_driver_api = {
 		.base = (PWM_Type *)DT_REG_ADDR(DT_INST_PARENT(n)),	  \
 		.index = DT_INST_PROP(n, index),			  \
 		.mode = kPWM_EdgeAligned,				  \
-		.prescale = kPWM_Prescale_Divide_128,			  \
+		.prescale = _CONCAT(kPWM_Prescale_Divide_, DT_INST_PROP(n, nxp_prescaler)),\
```

Comment:
`DT_INST_ENUM_IDX` work better?

Reply:
Not really, this approach is bulletproof and we are sure that we set the correct value based on the `_pwm_clock_prescale` enum at the build time. There is a couple of implementation in the codebase that successfully uses this solution, see:

<pre>
zephyr/drivers/watchdog/wdt_mcux_wdog32.c:#define TO_WDOG32_CLK_SRC(val) _DO_CONCAT(kWDOG32_ClockSource, val)
zephyr/drivers/watchdog/wdt_mcux_wdog32.c:#define TO_WDOG32_CLK_DIV(val) _DO_CONCAT(kWDOG32_ClockPrescalerDivide, val)
zephyr/drivers/counter/counter_mcux_lptmr.c:#define TO_LPTMR_CLK_SEL(val) _DO_CONCAT(kLPTMR_PrescalerClock_, val)
zephyr/drivers/counter/counter_mcux_lptmr.c:#define TO_LPTMR_PIN_SEL(val) _DO_CONCAT(kLPTMR_PinSelectInput_, val)
zephyr/drivers/counter/counter_mcux_lptmr.c:#define TO_LPTMR_PRESCALER(val) _DO_CONCAT(LPTMR_PRESCALER_, val)
zephyr/drivers/counter/counter_mcux_lptmr.c:#define TO_LPTMR_GLITCH(val) _DO_CONCAT(LPTMR_GLITCH_, val)
zephyr/drivers/pwm/pwm_mcux_pwt.c:#define TO_PWT_PRESCALE_DIVIDE(val) _DO_CONCAT(kPWT_Prescale_Divide_, val)
zephyr/drivers/pwm/pwm_mcux_ftm.c:#define TO_FTM_PRESCALE_DIVIDE(val) _DO_CONCAT(kFTM_Prescale_Divide_, val)
zephyr/drivers/adc/adc_mcux_lpadc.c:	_DO_CONCAT(kLPADC_ReferenceVoltageAlt, val)
zephyr/drivers/adc/adc_mcux_lpadc.c:	_DO_CONCAT(kLPADC_ConversionAverage, val)
zephyr/drivers/adc/adc_mcux_lpadc.c:	_DO_CONCAT(kLPADC_PowerLevelAlt, val)
zephyr/drivers/adc/adc_mcux_adc12.c:#define TO_ADC12_CLOCK_SRC(val) _DO_CONCAT(kADC12_ClockSourceAlt, val)
zephyr/drivers/adc/adc_mcux_adc12.c:#define TO_ADC12_CLOCK_DIV(val) _DO_CONCAT(kADC12_ClockDivider, val)
zephyr/drivers/adc/adc_mcux_12b1msps_sar.c:#define TO_RT_ADC_CLOCK_DIV(val) _DO_CONCAT(kADC_ClockDriver, val)
zephyr/drivers/timer/mcux_lptmr_timer.c:#define TO_LPTMR_PRESCALER(val) _DO_CONCAT(LPTMR_PRESCALER_, val)
zephyr/drivers/timer/mcux_lptmr_timer.c:#define TO_LPTMR_CLK_SEL(val) _DO_CONCAT(kLPTMR_PrescalerClock_, val)
zephyr/drivers/dac/dac_mcux_dac.c:	_DO_CONCAT(kDAC_ReferenceVoltageSourceVref, val)
zephyr/drivers/dac/dac_mcux_dac32.c:	_DO_CONCAT(kDAC32_ReferenceVoltageSourceVref, val)
</pre>